### ISSUE TITLE
Garbled text (systematic per-range character shift) on Type0 fonts with ToUnicode present — extracted silently, never flagged for OCR

### ISSUE THREAD

**Issue by abimaelmartell** [firecrawl/pdf-inspector#118]
## Summary

`text_simple__att10k.pdf` (ParseBench eval corpus) extracts as consistent mojibake: every character comes out shifted by a constant per character class (lowercase −15, uppercase −11, digits/punctuation by other fixed offsets). The document is classified TEXT-BASED, `ToUnicode` is present (`toUni=true`), no decode error is raised, and **neither `is_garbage_text` nor `detect_encoding_issues` flags it** — so `needs_ocr` stays false and the garbage is served silently. Downstream (fire-pdf / ParseBench) this doc fails every content and formatting rule (28 `is_underline` rules alone) despite structurally perfect extraction — even the `<u>` underline spans emit correctly around the garbled text.

## Reproduction

File: `text_simple__att10k.pdf` from the ParseBench dataset (HuggingFace `llamaindex/ParseBench`, `docs/text/text_simple__att10k.pdf`). Local copy: `~/Code/fire-pdf/evals/parsebench/vendor/ParseBench/data/docs/text/text_simple__att10k.pdf` (sha256 prefix `8b9aecf13c2eeee8`). It is an AT&T 10-K exhibit index — plenty of ground truth text to compare against.

```bash
cargo run --release --bin pdf2md -- text_simple__att10k.pdf | head -5
# 3FFQ&
# -'V 8VceZWZTReVW9VdZXReZdhZeYcVdaVTeeHVcZVd8EcVWVccVUHeT:iYZSZe-(,e2'WZ]V ;VScfRcj2&,*,*$ ...
```

`detect-pdf` says `Type: TEXT-BASED (extractable text)`; `extractPagesMarkdown` returns `pagesNeedingOcr: []`, `ocrReasonsByPage: []`.

## Evidence of the shift pattern

Extracted vs actual (from the rendered page):

| extracted | actual |
|---|---|
| `8VceZWZTReV` | `Certificate` |
| `9VdZXReZ...` | `Designati...` |
| `:iYZSZe` | `Exhibit` |
| `;VScfRcj2&,*,*$` | `February 18, 2020)` (approx.) |

Consistent offsets: lowercase `char − 15` (`e`→`V`, `r`→`c`, `t`→`e`, `i`→`Z`, `f`→`W`, `a`→`R`), uppercase `char − 11` (`C`→`8`, `D`→`9`, `E`→`:`, `F`→`;`), digits/punctuation shifted by their own fixed amounts. This is the signature of a **ToUnicode `bfrange` being misapplied (or authored broken)** — each contiguous range maps with a constant delta, and the deltas differ per range.

## Font table

```
font G1  sub=Type0  base=AAAAAB+font000000002f60c799  toUni=true  enc=Identity-H  cid=true
font G2  sub=Type0  base=AAAAAC+font000000002f60c799  toUni=true  enc=Identity-H  cid=true
```

Subset-tagged synthetic font names, Identity-H, CID, with ToUnicode CMaps that are either lying or being parsed with an off-by-range error.

## Suggested angles

1. **Decode**: check `tounicode.rs` bfrange handling against this file's CMaps (dump the parsed ranges vs the raw CMap stream). If the CMap itself is authored broken (producer bug — some webtool exports do this), the embedded TrueType `cmap` fallback (`get_font_file2_obj_num` path) or glyph-name recovery may decode correctly and `choose_best_cmap_decode` should prefer it. Worth verifying which decode source currently wins for G1/G2.
2. **Detection (independent fix, cheap)**: this text defeats `is_garbage_text`/`detect_encoding_issues` because it is 100% printable ASCII with word-like token lengths. A cheap discriminator that catches it: English-letter frequency / vowel ratio (the shifted text has almost no vowels — `8VceZWZTReVW9VdZXReZdhZeY` has 1 vowel in 25 chars) or dictionary-hit rate on 3+ letter tokens. Flagging `needs_ocr` + `suspected_garbled_text` for such pages would stop serving silent garbage even before the decode fix, and fire-pdf would route the page to GPU OCR which reads it fine.

## Impact

- Any consumer of this font-producer class gets ciphertext with no error signal (the dangerous part — it *looks* like a successful extraction).
- ParseBench: the doc fails all content + formatting rules (28 is_underline, plus is_bold/content); measured during fire-pdf ENG-5015.

Found during fire-pdf ENG-5015 (formatting round 2b) — see the per-doc underline analysis there for context.
